src/message/messagesReducer.js

```diff
@@ -142,7 +142,7 @@ export default (
         }
 
         const historyEntry = (() => {
-          if (action.edit_timestamp === undefined || action.user_id === undefined) {
+          if (action.edit_timestamp == null || action.user_id == null) {
```

Comment:
In 48a9bd436, to make the app aware of these "stealth" edits, we started by making the type for `action.user_id` optional:

```
user_id?: UserId,
```

Should we change that to `user_id: UserId | null` now?

Reply:
Yeah, I waffled on that. (Or rather `user_id?: UserId | null`.) It feels a bit premature as we haven't actually decided that API change yet. OTOH it would be nice to have the type-checker point out where the change in logic is needed.

My current thinking is we'll leave the type update to be done when there's a docs update and new feature level to point to. The new feature level also provides a convenient prompt for us to make the update.

Reply:
... And I waffled back. I'll include that change, with a comment to make clear it's a proposed future.

Reply:
> (Or rather `user_id?: UserId | null`.)

I was thinking `user_id: UserId | null`, because if we choose `null`, we won't also be using "absent" to mean the same thing (or anything else). And I think we wouldn't have some servers doing it one way, and some the other way; we'd settle on something before making a server release.

But this way is fine. It'll fit either way, and as you say we can come back and fix it when the server release is actually made. …Mm, I think that's probably what you meant with the comment `TODO(server-5.0): Update this and/or simplify.`.